src/librustc_parse/parser/diagnostics.rs

```diff
@@ -225,8 +225,21 @@ impl<'a> Parser<'a> {
             self.token.span,
             &format!("expected identifier, found {}", self.this_token_descr()),
         );
+        let valid_follow = &[
+            TokenKind::Eq,
+            TokenKind::Colon,
+            TokenKind::Comma,
+            TokenKind::Semi,
+            TokenKind::ModSep,
+            TokenKind::OpenDelim(token::DelimToken::Brace),
+            TokenKind::OpenDelim(token::DelimToken::Paren),
+            TokenKind::CloseDelim(token::DelimToken::Brace),
+            TokenKind::CloseDelim(token::DelimToken::Paren),
+        ];
```

Comment:
Code like this is pure technical debt, and my preferred solution would be to remove the diagnostic (https://github.com/rust-lang/rust/issues/66126#issuecomment-549956637), especially given that raw identifiers are a compatibility feature that should never be *recommended* in general.

Reply:
I think that with this new restriction we'll only be presenting this suggestion in the very rare cases where the _intent_ was there.